galley/pkg/config/processing/snapshotter/snapshotter.go

```diff
@@ -170,10 +170,12 @@ func (sg *snapshotGroup) onSync(c *coll.Instance) {
 	if !sg.synced[c] {
 		sg.remaining--
 		sg.synced[c] = true
+		scope.Processing.Debugf("sg.onSync: %v fully synced, %d remaining", c.Name(), sg.remaining)
```

Comment:
FYI: This will conflict with the race-fix PR that I have

Reply:
Rebased.